cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesExport.java

```diff
@@ -0,0 +1,408 @@
+                    Complex v2 = complexVoltage(dl.getR(), dl.getX(), dl.getG(), dl.getB(), b.getV(), b.getAngle(), dl.getP0(), dl.getQ0());
+                    writeVoltage(topologicalNode.get(), v2.abs(), Math.toDegrees(v2.getArgument()), writer);
+                } else {
+                    writeVoltage(topologicalNode.get(), 0.0, 0.0, writer);
+                }
+            }
+        }
+        // Voltages at inner nodes of Tie Lines
+        // (boundary nodes that have been left inside CGM)
+        for (Line l : network.getLines()) {
+            if (!l.isTieLine()) {
+                continue;
+            }
+            TieLine tieLine = (TieLine) l;
+            // FIXME(Luma) Obtain voltage at inner node
+            LOG.error("Must export topologicalNode voltage for boundary Tie Line {}", tieLine);
+        }
+    }
+
+    private static void writeVoltage(String topologicalNode, double v, double angle, XMLStreamWriter writer) throws XMLStreamException {
+        writer.writeStartElement(CIM_NAMESPACE, "SvVoltage");
+        writer.writeAttribute(RDF_NAMESPACE, ID, getUniqueId());
+        writer.writeStartElement(CIM_NAMESPACE, SV_VOLTAGE_ANGLE);
+        writer.writeCharacters(fs(angle));
+        writer.writeEndElement();
+        writer.writeStartElement(CIM_NAMESPACE, SV_VOLTAGE_V);
+        writer.writeCharacters(fs(v));
+        writer.writeEndElement();
+        writer.writeEmptyElement(CIM_NAMESPACE, SV_VOLTAGE_TOPOLOGICAL_NODE);
+        writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, "#" + topologicalNode);
+        writer.writeEndElement();
+    }
+
+    private static void writePowerFlows(Network network, XMLStreamWriter writer) {
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getLoadStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getGeneratorStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getShuntCompensatorStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getStaticVarCompensatorStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getBatteryStream);
+
+        network.getDanglingLineStream().forEach(dl -> dl.getAliasFromType(CGMES_PREFIX_ALIAS + "Terminal_Boundary")
+                .ifPresent(terminal -> writePowerFlow(terminal, dl.getP0(), dl.getQ0(), writer)));
+        // TODO what about flows of dl's generations?
+    }
+
+    private static <I extends Injection<I>> void writeInjectionPowerFlowToCgmes(Network network, XMLStreamWriter writer, Function<Network, Stream<I>> getInjectionStream) {
+        getInjectionStream.apply(network).forEach(i -> writePowerFlow(i.getTerminal(), writer));
+    }
+
+    private static void writePowerFlow(Terminal terminal, XMLStreamWriter writer) {
+        String cgmesTerminal = ((Connectable<?>) terminal.getConnectable()).getAliasFromType(CGMES_PREFIX_ALIAS + CgmesNames.TERMINAL1).orElse(null);
```

Comment:
And the TERMINAL2 ?

Reply:
This method is only used for injections which only have one terminal, which is why terminal2 is never called. I'm going to check what is done for branches/transformers (do you know @zamarrenolm @marqueslanauja ?)

Reply:
usually the flows for branches are not put in the SV output (they can always be easily recalculated, given the voltages and the impedances). We could export also these values, could be configurable.

Reply:
I'm adding a `TODO` about this but I think we can not export them at first